File: frontend/src/types/knife.ts
export interface KnifeListItem {
  id: number;
  name: string;
  price: number;
  mainImageUrl: string | null;
  createdAt: string;
}

export interface KnifeImage {
  id: number;
  url: string;
  isMain: boolean;
  sortOrder: number;
}

export interface KnifeDetail {
  id: number;
  name: string;
  price: number;
  steel: string;
  handle: string;
  sheath: string;
  totalLength: number;
  workingLength: number;
  maxWidth: number;
  thickness: number;
  createdAt: string;
  images: KnifeImage[];
}

export interface KnifeFormData {
  name: string;
  price: number;
  steel: string;
  handle: string;
  sheath: string;
  totalLength: number;
  workingLength: number;
  maxWidth: number;
  thickness: number;
  createdAt: string;
}

export interface LoginResponse {
  token: string;
  expiresAt: string;
}

File: backend/NozeStrona.Api/Models/Knife.cs
namespace NozeStrona.Api.Models;

public class Knife
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string Steel { get; set; } = string.Empty;
    public string Handle { get; set; } = string.Empty;
    public string Sheath { get; set; } = "BRAK";
    public decimal TotalLength { get; set; }
    public decimal WorkingLength { get; set; }
    public decimal MaxWidth { get; set; }
    public decimal Thickness { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<KnifeImage> Images { get; set; } = new List<KnifeImage>();
}
